src/SprykerEco/Service/ComputopApi/Model/Converter/ComputopApiConverter.php

```diff
@@ -0,0 +1,185 @@
+use SprykerEco\Shared\ComputopApi\ComputopApiConfig;
+use SprykerEco\Shared\ComputopApi\Config\ComputopApiConfig as ComputopApiConstants;
+
+class ComputopApiConverter extends AbstractComputopApi implements ComputopApiConverterInterface
+{
+    /**
+     * @param array $decryptedArray
+     * @param string $method
+     *
+     * @return \Generated\Shared\Transfer\ComputopApiResponseHeaderTransfer
+     */
+    public function extractHeader(array $decryptedArray, $method)
+    {
+        $decryptedArray = $this->formatResponseArray($decryptedArray);
+        $this->checkDecryptedResponse($decryptedArray);
+
+        $header = new ComputopApiResponseHeaderTransfer();
+        $header->fromArray($decryptedArray, true);
+        $header->setMId($this->getResponseValue($decryptedArray, ComputopApiConstants::MERCHANT_ID_SHORT));
+        $header->setTransId($this->getResponseValue($decryptedArray, ComputopApiConstants::TRANS_ID));
+        $header->setPayId($this->getResponseValue($decryptedArray, ComputopApiConstants::PAY_ID));
+        //optional
+        $header->setMac($this->getResponseValue($decryptedArray, ComputopApiConstants::MAC));
+        $header->setXId($this->getResponseValue($decryptedArray, ComputopApiConstants::X_ID));
+
+        $header->setIsSuccess(
+            $header->getStatus() === ComputopApiConfig::SUCCESS_STATUS ||
+            $header->getStatus() === ComputopApiConfig::SUCCESS_OK_STATUS
+        );
+        $header->setMethod($method);
+
+        return $header;
+    }
+
+    /**
+     * @param array $responseArray
+     * @param string $key
+     *
+     * @return null|string
+     */
+    public function getResponseValue(array $responseArray, $key)
+    {
+        if (isset($responseArray[$this->formatKey($key)])) {
+            return $responseArray[$this->formatKey($key)];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param string $decryptedString
+     *
+     * @return array
+     */
+    public function getResponseDecryptedArray($decryptedString)
+    {
+        $decryptedArray = [];
+        $decryptedSubArray = explode(self::DATA_SEPARATOR, $decryptedString);
+        foreach ($decryptedSubArray as $value) {
+            $data = explode(self::DATA_SUB_SEPARATOR, $value);
+            $decryptedArray[array_shift($data)] = array_shift($data);
+        }
+
+        return $this->formatResponseArray($decryptedArray);
+    }
+
+    /**
+     * @param array $responseArray
+     *
+     * @throws \SprykerEco\Service\ComputopApi\Exception\ComputopApiConverterException
+     *
+     * @return void
+     */
+    public function checkEncryptedResponse(array $responseArray)
+    {
+        $keys = [
+            ComputopApiConstants::DATA,
+            ComputopApiConstants::LENGTH,
+        ];
+
+        if (!$this->existArrayKeys($keys, $responseArray)) {
+            throw new ComputopApiConverterException(
+                'Response does not have expected values. Please check Computop documentation.'
```

Comment:
This is too generic. If there are 70 values and only 1 of them is wrong, it will be super difficult to figure out the problem.

The `existArrayKeys` should return the exact list or at least the 1 value where it got stuck, and that should be displayed in the error message.